test/e2e/ingress.go

```diff
@@ -243,31 +374,68 @@ var _ = Describe("GCE L7 LoadBalancer Controller", func() {
 		framework.beforeEach()
 		client = framework.Client
 		ns = framework.Namespace.Name
-		Expect(waitForRCPodsRunning(client, "kube-system", "glbc")).NotTo(HaveOccurred())
-		Expect(checkLeakedResources()).NotTo(HaveOccurred())
+		// Scaled down the existing Ingress controller so it doesn't interfere with the test.
+		Expect(scaleKubeSystemRC(client, lbContainerName, 0)).NotTo(HaveOccurred())
```

Comment:
This seems like entirely the wrong approach to me; it disables the cluster's real ingress controller and tests functionality of one that isn't running in the cluster by default. An end user is not going to do that or even know that they can/should. We should be testing functionality of the system-managed controller not a mock.